xs/src/libslic3r/SimplePrint.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef slic3r_SimplePrint_hpp_
+#define slic3r_SimplePrint_hpp_
+
+#include "libslic3r.h"
+#include "Point.hpp"
+#include "Print.hpp"
+
+namespace Slic3r {
+
+class SimplePrint {
+    public:
+    bool center{true};
+    std::function<void(int, const std::string&)> status_cb {nullptr};
+    
+    bool apply_config(DynamicPrintConfig config) { return this->_print.apply_config(config); }
+    double total_used_filament() const { return this->_print.total_used_filament; }
+    double total_extruded_volume() const { return this->_print.total_extruded_volume; }
+    void set_model(const Model &model);
+    void export_gcode(std::string outfile);
```

Comment:
I'm wondering if a regression test is feasible here, given a fixed initial config. Could we split this method so it outputs a string and tests some of the CLI-only parameters via GCodeReader? I think that would be resilient enough from changes elsewhere in Slic3r.